### ISSUE TITLE
`yarn test` on root folder fails

### ISSUE THREAD

**Issue by Turbo87** [embroider-build/embroider#77]
```
$ yarn test
yarn run v1.15.2
$ ./scripts/test.sh
Academic tradition requires you to cite works you base your article on.
If you use programs that use GNU Parallel to process data for an article in a
scientific publication, please cite:

  O. Tange (2018): GNU Parallel 2018, Mar 2018, ISBN 9781387509881,
  DOI https://doi.org/10.5281/zenodo.1146014

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

More about funding GNU Parallel and the citation notice:
https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice

To silence this citation notice: run 'parallel --citation' once.

  node	$ qunit packages/*/tests
  node	info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
  node	Some tests failed
  node	No files were found matching: packages/compat/tests, packages/core/tests, packages/macros/tests
  node	error Command failed with exit code 1.
  macro-addon	$ ember test --test-port=0
  macro-addon	info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
  macro-addon	Some tests failed
  macro-addon	The "id" argument must be of type string. Received type undefined
  macro-addon
  macro-addon
  macro-addon	Stack Trace and Error Report: /var/folders/6y/_8qpkt114jzb442txt_9w_2c0000gn/T/error.dump.4bc92e4000af482ae5a032babbf9164b.log
  macro-addon	An error occurred in the constructor for macro-sample-addon at /Users/tbieniek/Code/embroider/test-packages/macro-sample-addon
  macro-addon
  macro-addon	error Command failed with exit code 1.
  macro	$ ember test --test-port=0
  macro	info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
  macro	Some tests failed
  macro	The "id" argument must be of type string. Received type undefined
  macro
  macro
  macro	Stack Trace and Error Report: /var/folders/6y/_8qpkt114jzb442txt_9w_2c0000gn/T/error.dump.716ee68652fa25e2d9caa7ada1b81af3.log
  macro	error Command failed with exit code 1.
  macro-classic	$ ember test --test-port=0
  macro-classic	info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
  macro-classic	Some tests failed
  macro-classic	The "id" argument must be of type string. Received type undefined
  macro-classic
  macro-classic
  macro-classic	Stack Trace and Error Report: /var/folders/6y/_8qpkt114jzb442txt_9w_2c0000gn/T/error.dump.592e59567a37d9bcc2ca5cf97237b986.log
  macro-classic	error Command failed with exit code 1.
  macro-addon-classic	$ ember test --test-port=0
  macro-addon-classic	info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
  macro-addon-classic	Some tests failed
  macro-addon-classic	The "id" argument must be of type string. Received type undefined
  macro-addon-classic
  macro-addon-classic
  macro-addon-classic	Stack Trace and Error Report: /var/folders/6y/_8qpkt114jzb442txt_9w_2c0000gn/T/error.dump.6332bf1b2de093906ee39490bfef0f33.log
  macro-addon-classic	An error occurred in the constructor for macro-sample-addon at /Users/tbieniek/Code/embroider/test-packages/macro-sample-addon
  macro-addon-classic
  macro-addon-classic	error Command failed with exit code 1.
  static-app	$ ember test
  static-app	info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
  static-app	Some tests failed
  static-app	The "id" argument must be of type string. Received type undefined
  static-app
  static-app
  static-app	Stack Trace and Error Report: /var/folders/6y/_8qpkt114jzb442txt_9w_2c0000gn/T/error.dump.005231c39741a1570706a0af5db56c88.log
  static-app	error Command failed with exit code 1.
  static-app-classic	$ ember test
  static-app-classic	info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
  static-app-classic	Some tests failed
  static-app-classic	The "id" argument must be of type string. Received type undefined
  static-app-classic
  static-app-classic
  static-app-classic	Stack Trace and Error Report: /var/folders/6y/_8qpkt114jzb442txt_9w_2c0000gn/T/error.dump.5e1b6f232aa6497e0f9bdff291993f38.log
  static-app-classic	error Command failed with exit code 1.
Some tests failed
error Command failed with exit code 7.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by Turbo87** [comment]
```
  - stack: TypeError [ERR_INVALID_ARG_TYPE]: The "id" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:125:11)
    at Module.require (internal/modules/cjs/loader.js:662:3)
    at require (internal/modules/cjs/helpers.js:20:18)
    at PackageInfo.getAddonConstructor (/Users/tbieniek/Code/embroider/node_modules/ember-cli/lib/models/package-info-cache/package-info.js:380:18)
    at graph.each (/Users/tbieniek/Code/embroider/node_modules/ember-cli/lib/models/instantiate-addons.js:70:38)
    at Vertices.each (/Users/tbieniek/Code/embroider/node_modules/dag-map/dag-map.umd.js:197:13)
    at Vertices.walk (/Users/tbieniek/Code/embroider/node_modules/dag-map/dag-map.umd.js:125:14)
    at DAG.each (/Users/tbieniek/Code/embroider/node_modules/dag-map/dag-map.umd.js:68:24)
    at instantiateAddons (/Users/tbieniek/Code/embroider/node_modules/ember-cli/lib/models/instantiate-addons.js:53:9)
    at Project.initializeAddons (/Users/tbieniek/Code/embroider/node_modules/ember-cli/lib/models/project.js:471:19)
```

**Comment by Turbo87** [comment]
looks like I forgot/didn't know about running `yarn compile` before running the test suite...

**Comment by Turbo87** [comment]
this is still failing though :-/

```
  static-app	not ok 1 [undefined ms] - Error
  static-app	    ---
  static-app	        message: >
  static-app	            listen EADDRINUSE: address already in use :::7357
  static-app	    ...
  static-app
  static-app	1..1
  static-app	# tests 1
  static-app	# pass  0
  static-app	# skip  0
  static-app	# fail  1
```

**Comment by Turbo87** [comment]
fixed by the documentation update in  #91

**Comment by jfrux** [comment]
@Turbo87 Hey! I'm getting this error in a completely unrelated project.  It's happening when running a host app with ember engine.  

`The "id" argument must be of type string. Received type undefined`

Do you recall what it was that actually caused the error itself?
